Expressions in action payloads (#1943)
oasis-tcs/odata-specs: odata-json-format/16 Bound Function.md

appropriate to the action.
Stream typed parameter values are represented following the same rules as inlined [stream properties](#StreamProperty).

Alternatively, values of non-binding parameters MAY be specified as common expressions
[OData-URL, section 5.1.1](#ODataURL) that the service evaluates relative to the

> action. They are then encoded as a name/value

Author: Relative to what are expressions in function invocations evaluated?
```
/service/Employees(23)/CheckAbsence(Employee=@colleague,Date=@date)
?@colleague=Manager
&@date=now() add duration'P1D'
```

Reviewer: I'd say the base URL in this example is scheme plus host (and port) plus `/service/Employees(23)/` - note the trailing slash.

Reviewer: Seems to be covered now: anything I missed?

appropriate to the action.
Stream typed parameter values are represented following the same rules as inlined [stream properties](#StreamProperty).

Alternatively, values of non-binding parameters MAY be specified as common expressions
[OData-URL, section 5.1.1](#ODataURL) that the service evaluates relative to the
action. They are then encoded as a name/value
pair where the name is the name of the parameter followed by `@expressionUrl` and
the value is the common expression. As the following example demonstrates, even
non-transient entities can be passed as non-binding action parameters in this way.

::: example
Example ##ex: An employee requests leave from their manager for the next two weeks.
```json
POST /service/Employees(23)/RequestLeave
Host: host
Content-Type: application/json

{
  "StartDate@expressionUrl": "now()",
  "EndDate@expressionUrl": "now() add duration'P14D'",

>   "Approver@expressionUrl": "$root/Employees(23)/Manager"

Reviewer: We should add an example where relative resolution matters, and explain how it is resolved.

appropriate to the action.
Stream typed parameter values are represented following the same rules as inlined [stream properties](#StreamProperty).

Alternatively, values of non-binding parameters MAY be specified as common expressions
[OData-URL, section 5.1.1](#ODataURL). In the case of a bound action
these MAY contain path expressions [OData-URL, section 5.1.1.15](#ODataURL), which

> the service evaluates on the binding parameter. Such parameters are encoded as a name/value
> pair where the name is the name of the parameter followed by `@expressionUrl` and

Reviewer: ```suggestion
the service evaluates on the binding parameter value. Such parameters are encoded as name/value
pairs where the name is the name of the parameter followed by `@expressionUrl` and
```

the value is the common expression. As the following example demonstrates, even
non-transient entities can be passed as non-binding action parameters in this way.

::: example
Example ##ex: An employee requests leave from their manager for the next two weeks:
```json
POST /service/Employees(23)/self.RequestLeave
Host: host
Content-Type: application/json

{
  "StartDate@expressionUrl": "now()",
  "EndDate@expressionUrl": "now() add duration'P14D'",
  "Approver@expressionUrl": "Manager"
}
```

> The expression `Manager` is evaluated on the binding parameter `Employees(23)`.

Reviewer: ```suggestion
The expression `Manager` is evaluated on the binding parameter value `Employees(23)`.
```

appropriate to the action.
Stream typed parameter values are represented following the same rules as inlined [stream properties](#StreamProperty).

Alternatively, values of non-binding parameters MAY be specified as common expressions
[OData-URL, section 5.1.1](#ODataURL). In the case of a bound action
these MAY contain path expressions [OData-URL, section 5.1.1.15](#ODataURL), which
the service evaluates on the binding parameter value. Such parameters are encoded as name/value
pairs where the name is the name of the parameter followed by `@expressionUrl` and
the value is the common expression. As the following example demonstrates,
non-transient entities can be passed as non-binding action parameters through a
resource path in this way.

::: example
Example ##ex: An employee requests leave from their manager for the next two weeks:
```json
POST /service/Employees(23)/self.RequestLeave
Host: host
Content-Type: application/json

{

>   "StartDate@expressionUrl": "now()",

Reviewer: ```suggestion
  "StartDate@expression": "now()",
```
Rename throughout
